staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -396,8 +396,10 @@ func interfaceFromUnstructured(sv, dv reflect.Value) error {
 func (c *converterImpl) ToUnstructured(obj interface{}) (map[string]interface{}, error) {
 	var u map[string]interface{}
 	var err error
-	if unstr, ok := obj.(runtime.Unstructured); ok {
-		u = DeepCopyJSON(unstr.UnstructuredContent())
+	ro, ro_ok := obj.(runtime.Object)
+	if _, ok := obj.(runtime.Unstructured); ok && ro_ok {
+		// UnstructuredContent() mutates the object so we need to make a copy first
+		u = ro.DeepCopyObject().(runtime.Unstructured).UnstructuredContent()
```

Comment:
`u = ro.DeepCopy().UnstructuredContent()`

Reply:
`u` is `map[string]interface{}` :)

Reply:
`runtime.Unstructured` does not implement `runtime.Object`

Reply:
Maybe it should?

Reply:
It makes sense to me. Add this change to this PR?

Reply:
it's a twoliner? then why not in an extra commit.